tests/foreman/api/test_location.py

```diff
@@ -0,0 +1,210 @@
+"""Unit tests for the ``locations`` paths.
+
+A full API reference for locations can be found here:
+http://theforeman.org/api/apidoc/v2/locations.html
+
+"""
+from ddt import ddt
+from fauxfactory import gen_string
+from nailgun import entities
+from random import randint
+from robottelo.common.decorators import data
+from robottelo.test import APITestCase
+# (too-many-public-methods) pylint:disable=R0904
+
+
+@ddt
+class LocationTestCase(APITestCase):
+    """Tests for the ``locations`` path."""
+    # TODO Add coverage for media, smart_proxy, realms once they implemented
+
+    @data(
+        gen_string('alphanumeric', randint(1, 255)),
+        gen_string('alpha', randint(1, 255)),
+        gen_string('cjk', randint(1, 85)),
+        gen_string('latin1', randint(1, 255)),
+        gen_string('numeric', randint(1, 255)),
+        gen_string('utf8', randint(1, 85)),
+    )
+    def test_create_location_with_different_names(self, name):
+        """@Test: Create new locations using different inputs as a name
+
+        @Assert: Location created successfully and has expected and correct
+        name
+
+        @Feature: Location
+
+        """
+        location = entities.Location(name=name).create()
+        self.assertEqual(location.name, name)
+
+    def test_create_location_with_description(self):
+        """@Test: Create new location with custom description
+
+        @Assert: Location created successfully and has expected and correct
+        description
+
+        @Feature: Location
+
+        """
+        description = gen_string('utf8')
+        location = entities.Location(description=description).create()
+        self.assertEqual(location.description, description)
+
+    def test_create_location_with_user(self):
+        """@Test: Create new location with assigned user to it
+
+        @Assert: Location created successfully and has correct user assigned to
+        it with expected login name
+
+        @Feature: Location
+
+        """
+        user = entities.User().create()
+        location = entities.Location(user=[user.id]).create()
+        self.assertEqual(location.user[0].id, user.id)
+        self.assertEqual(location.read_json()['users'][0]['login'], user.login)
+
+    def test_create_location_with_comp_resource_libvirt(self):
+        """@Test: Create new location with Libvirt compute resource assigned to
+        it
+
+        @Assert: Location created successfully and has correct Libvirt compute
+        resource assigned to it
+
+        @Feature: Location
+
+        """
+        test_cr = entities.LibvirtComputeResource().create()
+        location = entities.Location(compute_resource=[test_cr.id]).create()
```

Comment:
This works fine, but it's even more idiomatic to write `compute_resource=[test_cr]`.